src/libraries/MerchantLogic.sol

```diff
@@ -7,25 +7,27 @@ library MerchantLogic {
     /// @notice Emitted when a merchant updates their configuration
     /// @param merchant Address of the merchant
     /// @param payoutToken Target token address for payouts
-    /// @param payoutAddress Address that will receive payouts
-    /// @param fulfillerPercentage Percentage of payout that goes to fulfiller in 6 decimals precision
+    /// @param payoutAddresses Array of payout addresses
+    /// @param payoutPercentages Array of payout percentages
     event MerchantConfigSet(
-        address indexed merchant, address payoutToken, address payoutAddress, uint32 fulfillerPercentage
+        address indexed merchant, address payoutToken, address[] payoutAddresses, uint32[] payoutPercentages
     );
 
-    /// @notice Error thrown when fulfillerPercentage is greater than 100% (1e6 precision)
-    error InvalidFulfillerPercentage();
-
-    /// @notice Error thrown when payoutAddress is zero
+    /// @notice Error thrown when payout address is zero
     error PayoutAddressCannotBeZero();
 
-    uint32 public constant PERCENTAGE_PRECISION = 1e6;
-    uint32 public constant MAX_PERCENTAGE = 100 * PERCENTAGE_PRECISION;
+    /// @notice Error thrown when payout percentages don't sum to 100%
+    error InvalidPayoutPercentages();
+
+    /// @notice Error thrown when no payout recipients are provided
+    error NoPayoutRecipients();
+
+    uint32 public constant TOTAL_PERCENTAGE = 100;
```

Comment:
Consider using uint8. Or I would really recommend to use few digits precision to enable fees used cases as most fees are set to be less than 1%